core/src/main/java/org/testcontainers/containers/wait/HostPortWaitStrategy.java

```diff
@@ -23,78 +25,45 @@
 @Slf4j
 public class HostPortWaitStrategy extends GenericContainer.AbstractWaitStrategy {
 
-    private static final String SUCCESS_MARKER = "TESTCONTAINERS_SUCCESS";
     @Override
     protected void waitUntilReady() {
-        final Integer port = getLivenessCheckPort();
-        if (null == port) {
-            log.debug("Liveness check port of {} is empty. Not waiting.", container.getContainerName());
+        final List<Integer> externalLivenessCheckPorts = getLivenessCheckPorts();
+        if (null == externalLivenessCheckPorts || externalLivenessCheckPorts.isEmpty()) {
```

Comment:
since `getLivenessCheckPorts` returns a collection, it would be nice to mark it as `@NotNull` to avoid such checks and make the contract more strict (nullable collection doesn't make a lot of sense anyway :) )

Reply:
Good idea!